gribscan/gribscan.py

```diff
@@ -227,6 +227,14 @@ def get_time_offset(gribmessage, lean_towards="end"):
             offset += int(gribmessage["P1"]) * unit
         elif timeRangeIndicator == 1:
             pass
+        elif timeRangeIndicator in [2, 4]:
+            # Product with a valid time ranging between reference time + P1 and
+            # reference time + P2
+            unit = time_range_units[
+                int(gribmessage.get("indicatorOfUnitOfTimeRange", 255))
+            ]
+            offset += int(gribmessage["P1"]) * unit
+
```

Comment:
This section should probably check the `lean_towards` argument, which is either `"start"` or `"end"` and determines if the reference time point of some data valid in a time range should reflect the start of the range or the end of the range.

Reply:
@d70-t would you be able to help me with this? I don't know grib very well, so I don't quite understand how to do this...

Reply:
I don't know a whole lot about grib as well... but I guessed from your comment, that the data is to be interpreted as valid between `P1` and `P2`. But for a Dataset, we need one time value. And `lean_towards` should decide. Maybe something like:

```suggestion
        elif timeRangeIndicator in [2, 4]:
            # Product with a valid time ranging between reference time + P1 and
            # reference time + P2
            unit = time_range_units[
                int(gribmessage.get("indicatorOfUnitOfTimeRange", 255))
            ]
            if lean_towards == "end":
                offset += int(gribmessage["P2"]) * unit
            else:
                offset += int(gribmessage["P1"]) * unit

```
should do the trick?

Reply:
I would also try and summon @sbnielsen, he has a better overview of grib tables than I do - and for grib1 I am even less familiar with the tables.

@leifdenby, do I understand correctly that this is only related to giving the data a time point, and has nothing to do with how the data has been statistically processed? (eg. average, maximum, accumulated etc).

I think it may be best to handle the cases `timeRangeIndicator=[2,4]` separately, since `timeRangeIndicator=4` is considered valid at `reference time + P2` (https://codes.ecmwf.int/grib/format/grib1/ctable/5/) whereas for `timeRangeIndicator=2` is can be whatever in between `reference time + P1` and `reference time + P2`. So maybe the `lean_towards` argument is only necessary for `timeRangeIndicator=2`?

Reply:
I agree with your interpretation :+1: And that's also how we handled it for other `timeRangeIndicator`s.